### ISSUE TITLE
WithOpenApi cannot be used to set Summary and Description in OpenAPI documentation

### ISSUE THREAD

**Issue by marcominerva** [dotnet/aspnetcore#57713]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

If I try to set the `Summary` and `Description` properties of an endpoint using the `OpenApiOperation` object that is provided by `WithOpenApi`:

```csharp
builder.Services.AddOpenApi();

// ...

if (app.Environment.IsDevelopment())
{
    app.MapOpenApi();    
}

app.MapGet("/api/test-withopenapi", () =>
{
    return TypedResults.Ok();
})
.WithOpenApi(operation =>
{
    operation.Summary = "Setting the summary using OpenApiOperation object doesn't work";
    operation.Description = "Setting the description using OpenApiOperation object doesn't work";

    return operation;
});
```

The information isn't reflected in the OpenAPI documentation.

If, however, I use the `WithSummary` and `WithDescription` extension methods:

```csharp
app.MapGet("/api/test-with-extension-methods", () =>
{
    return TypedResults.Ok();
})
.WithOpenApi()
.WithSummary("Setting the summary using the WithSummary extension method works as expected")
.WithDescription("Setting the description using the WithDescription extension method works as expected");
```

Then, summary and description correctly appear in the file:

```json
"/api/test-with-extension-methods": {
  "get": {
      "tags": [
          "WithOpenApiIssue"
      ],
      "summary": "Setting the summary using the WithSummary extension method works as expected",
      "description": "Setting the description using the WithDescription extension method works as expected",
```

### Expected Behavior

Summary and description should be set even when using `WithOpenApi` (as it happens with **Swashbuckle.AspNetCore v6.7.3** and .NET 8.0).

### Steps To Reproduce

Minimal repro here: https://github.com/marcominerva/WithOpenApiNet9Issue

### .NET Version

9.0.100-preview.7.24407.12

### Anything else?

.NET SDK:
Version: 9.0.100-preview.7.24407.12
Commit: d672b8a045
Workload version: 9.0.100-manifests.baed1e37
MSBuild version: 17.12.0-preview-24374-02+48e81c6f1

Runtime Environment:
OS Name: Windows
OS Version: 10.0.17763
OS Platform: Windows
RID: win-x64
Base Path: C:\Program Files\dotnet\sdk\9.0.100-preview.7.24407.12\

.NET workloads installed:
Configured to use loose manifests when installing new manifests.
[aspire]
Installation Source: VS 17.12.35209.166, VS 17.11.35222.181
Manifest Version: 8.1.0/8.0.100
Manifest Path: C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.aspire\8.1.0\WorkloadManifest.json
Install Type: FileBased

[maccatalyst]
Installation Source: VS 17.11.35222.181
Manifest Version: 17.5.9231-net9-p7/9.0.100-preview.7
Manifest Path: C:\Program Files\dotnet\sdk-manifests\9.0.100-preview.7\microsoft.net.sdk.maccatalyst\17.5.9231-net9-p7\WorkloadManifest.json
Install Type: Msi

[ios]
Installation Source: VS 17.11.35222.181
Manifest Version: 17.5.9231-net9-p7/9.0.100-preview.7
Manifest Path: C:\Program Files\dotnet\sdk-manifests\9.0.100-preview.7\microsoft.net.sdk.ios\17.5.9231-net9-p7\WorkloadManifest.json
Install Type: Msi

[android]
Installation Source: VS 17.11.35222.181
Manifest Version: 35.0.0-preview.7.41/9.0.100-preview.7
Manifest Path: C:\Program Files\dotnet\sdk-manifests\9.0.100-preview.7\microsoft.net.sdk.android\35.0.0-preview.7.41\WorkloadManifest.json
Install Type: Msi

[maui-windows]
Installation Source: VS 17.11.35222.181
Manifest Version: 9.0.0-preview.7.24407.4/9.0.100-preview.7
Manifest Path: C:\Program Files\dotnet\sdk-manifests\9.0.100-preview.7\microsoft.net.sdk.maui\9.0.0-preview.7.24407.4\WorkloadManifest.json
Install Type: Msi

Host:
Version: 9.0.0-preview.7.24405.7
Architecture: x64
Commit: static

.NET SDKs installed:
6.0.100-preview.5.21302.13 [C:\Program Files\dotnet\sdk]
7.0.100-rc.1.22431.12 [C:\Program Files\dotnet\sdk]
8.0.400 [C:\Program Files\dotnet\sdk]
9.0.100-preview.7.24407.12 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
Microsoft.AspNetCore.App 6.0.0-preview.5.21301.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 6.0.32 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 6.0.33 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 7.0.0-rc.1.22427.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 7.0.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 8.0.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 8.0.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 9.0.0-preview.7.24406.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.NETCore.App 6.0.0-preview.5.21301.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 6.0.32 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 6.0.33 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 7.0.0-rc.1.22426.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 7.0.20 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 8.0.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 8.0.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 9.0.0-preview.7.24405.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.WindowsDesktop.App 6.0.0-preview.5.21301.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 6.0.32 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 6.0.33 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 7.0.0-rc.1.22427.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 7.0.20 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 8.0.7 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 8.0.8 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 9.0.0-preview.7.24405.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
x86 [C:\Program Files (x86)\dotnet]
registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
Not set

global.json file:
Not found

**Comment by martincostello** [comment]
I think these conversations are relevant here:

- https://github.com/dotnet/aspnetcore/issues/56585#issuecomment-2219447719
- https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues/2849

I _think_ the TL;DR is "this is expected as they're different implementations", but @captainsafia probably remembers better than I do.

**Comment by captainsafia** [comment]
Yep, https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues/2849 provides the conclusion about where we landed here. I also updated the docstrings on the `WithOpenApi` method to indicate they are only compatible with Swashbuckle.AspNetCore (since neither NSwag or Microsoft.AspNetCore.OpenApi) support it at the moment.

We haven't gone as far as obsoleting the API since Swashbuckle still uses it so the docstring will have to suffice for now.
